### PR TITLE
NEO-1452: Fix issue with Closable Chips in disabled state, both alone & inside Select

### PR DESCRIPTION
**Pull request by mrazian85**
[Link to updated components in Deploy Preview](https://deploy-preview-177--neo-react-library-storybook.netlify.app/?path=/story/components-select--required-in-form)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

As per [NEO-1452](https://jira.forge.avaya.com/browse/NEO-1452), this PR address an issue whereby the user is still able to interact with Closable Chips in disabled state using the mouse and keyboard.

The issue is resolved by means of the following:
- Passing the `disabled` state from the Select variant into the child Chip Components
- Setting the `disabled` attribute on the button element in the Closable Chip, along with the necessary styling (note that a separate ticket, [NEO-1549](https://jira.forge.avaya.com/browse/NEO-1549), has been created to add this styling to the CSS library).

I have also added a test for the Closable Chip in disabled state.

### CODE REVIEW

File: src/components/Chip/Chip.tsx [hunk 1]
```diff
@@ -67,6 +67,9 @@ export const Chip = ({
         <button
           className="neo-close neo-close--clear"
           aria-label={closeButtonAriaLabel}
+          // TO-DO: NEO-1549: Add the below styling rule in the CSS library
+          style={{ pointerEvents: disabled ? "none" : "initial" }}
```
Comment (joe-s-avaya) [inline src/components/Chip/Chip.tsx:71]: huh, I'm surprised that we need this `style`... oh well, makes sense, just surprising

**Review by joe-s-avaya — APPROVED** [review]
elegant, simple, tested, delicious

File: src/components/Chip/Chip.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/Chip/Chip.tsx:71]: This is much easier than a javascript solution.

**Review by yangAtSpoken — APPROVED** [review]
great

**Review by matthewAvaya — APPROVED** [review]

**Pull request merged**